crates/core/src/pipeline/placeholder.rs

```diff
@@ -0,0 +1,97 @@
+use anyhow::{bail, Result};
+use crate::variant::video::VideoVariant;
+use crate::variant::audio::AudioVariant;
+use ffmpeg_rs_raw::ffmpeg_sys_the_third::{
+    av_frame_alloc, av_frame_get_buffer, av_frame_free, av_get_sample_fmt, AVFrame, 
+    AVPixelFormat, AVSampleFormat
+};
+use ffmpeg_rs_raw::cstr;
+
+/// Placeholder frame generator for idle mode when stream disconnects
+pub struct PlaceholderGenerator;
+
+impl PlaceholderGenerator {
+    /// Generate a placeholder black video frame
+    pub unsafe fn generate_video_frame(
+        variant: &VideoVariant, 
+        stream_time_base: (i32, i32),
+        frame_count: u64
```

Comment:
This should be called frame_index instead frame_count makes it seem like it generates more than 1 frame